pkgs/build-support/node/build-npm-package/hooks/default.nix

```diff
@@ -1,10 +1,43 @@
-{ lib, makeSetupHook, nodejs, srcOnly, buildPackages, makeWrapper }:
+{ lib, stdenvNoCC, makeSetupHook, nodejs, srcOnly, buildPackages, makeWrapper }:
 
+let
+  patchedNode = stdenvNoCC.mkDerivation {
```

Comment:
I'll add the comment myself with specifics later, but we have to patch at all because the assumptions Node and npm make really don't do great for trying to reproducibly build all this stuff (like allowing packages to depend on their own node-gyp version instead of using the bundled version that we already patch to fix bad behavior)

The worst is the special handling of git deps in their pacote fetcher which straight up just forks a new `npm install` subprocess (from the pacote library!) which obviously is bad news for our tooling in regards to lockfile fixup and ensuring `patchShebangs` gets run for relevant scripts in `node_modules`

---

I want to get rid of the `patchedNode` here at some point and just conditionally enable some of the patched code paths in the main `nodejs` derivation when run in a build environment (e.g. detected by a `NIX_NODE_FIX_NPM` var we could patch in) and that way we don't have to have a difficult-to-override version of nodejs here and manually juggle adding it in front of `nativeBuildInputs` env var for example